website/docs/cli/commands/test/index.mdx

```diff
@@ -0,0 +1,368 @@
+
+### The `run.assert` block
+
+You can specify `assert` blocks inside your `run` block to test the state of your infrastructure after the
+`apply` or `plan` operation is complete. There is no theoretical limit to the number of blocks you can define.
+
+Each block requires the following two attributes:
+
+1. The `condition` is an [OpenTofu condition](/docs/language/expressions/custom-conditions#condition-expressions) which
+   should return `true` for the test to pass, `false` for the test to fail. The condition **must** reference a
+   resource, data source, variable, output or module from the main code, otherwise OpenTofu will refuse to run the test.
+2. The `error_message` is a string explaining what happened when the test fails.
+
+:::tip Example
+
+As a simple example, you can write an `assert` block as follows:
+
+<Tabs>
+    <TabItem label={"main.tftest.hcl"} value={"test"} default>
+        <CodeBlock language="hcl">{SimpleTest}</CodeBlock>
+    </TabItem>
+    <TabItem label={"main.tf"} value={"main"}>
+        <CodeBlock language="hcl">{SimpleMain}</CodeBlock>
+    </TabItem>
+</Tabs>
+
+:::
+
+Please note that conditions only let you perform basic checks on the current OpenTofu state and use OpenTofu functions.
+**You cannot define additional data sources directly in your test code.** To work around this limitation, you can use
+[the `module` block](#the-runmodule-block) in order to load a helper module.
+
+### The `run.module` block
+
+In some cases you may find that the tools provided in the
+[condition expression](/docs/language/expressions/custom-conditions#condition-expressions) are not enough to test
+if your code created the infrastructure correctly.
+
+You can use the `module` block to override the main module `tofu test` loads. This gives you the opportunity to create
+additional resources or data sources that you can use in your `assert` conditions.
+
+Its syntax is similar to loading modules in normal OpenTofu code:
+
+```hcl
+run "test" {
+  module {
+    source = "./some-module"
+  }
+}
+```
+
+The `module` block has the following two attributes:
+
+* The `source` attribute points to the directory of the module to load or any other [module source](/docs/language/modules/sources).
+* The `version` specifies the version of the module you want to use.
+
+:::warning Note
+
+You cannot pass parameters directly in the `module` block as you may be used to from the normal OpenTofu code. Use the
+[`variables` block](#the-variables-and-runvariables-blocks) to pass parameters to your module.
+
+:::
+
+:::tip Example
+
+In this example project the `main.tf` file creates a Docker container with an `nginx` image exposed on port 8080.
+The `main.tftest.hcl` file needs to test if the webserver actually starts properly, but it cannot do that without
+a helper module.
+
+To create the `http` data source, the `main.tftest.hcl` file loads the `test-helper` module. The test helper then loads
```

Comment:
How do you feel about not using the term `helper` here and instead using something that feels a little more test related. Maybe `test-harness` ?